src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -199,7 +269,7 @@ public virtual string CreateToken(SecurityTokenDescriptor tokenDescriptor)
             if (!payload.Any())
```

Comment:
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/1237 was opened as an empty payload should be allowed.

Reply:
Fixed. Also added in some tests.